### PR TITLE
Add convenience methods for content blocks

### PR DESCRIPTION
**Pull request by mjgiarlo**
Needed by https://github.com/projecthydra-labs/hybox/pull/82

### CODE REVIEW

**Comment by mjgiarlo** [comment]
Failure is unrelated to work done in this PR.

File: spec/models/content_block_spec.rb [hunk 1]
```diff
@@ -1,47 +1,98 @@
...
-    value: '<h1>Frodo Baggins</h1>',
-    created_at: 2.hours.ago)
-  }
-
-  let(:marketing) { described_class.create!(
-    name: ContentBlock::MARKETING,
-    value: '<h1>Marketing Text</h1>')
-  }
+  let(:bilbo) do
+    described_class.create(
```
Comment (awead) [inline spec/models/content_block_spec.rb]: I would use a factory here with `let!`. Ex:
``` ruby
let!(:bilbo) { create(:content_block, value: '<h1>Bilbo Baggins</h1>') }
let!(:frodo) { create(:content_block, value: '<h1>Frodo Baggins</h1>', :two_hours_ago) }
```

File: spec/models/content_block_spec.rb [hunk 2]
```diff
@@ -1,47 +1,98 @@
...
+      name: ContentBlock::ANNOUNCEMENT,
+      value: '<h1>Announcement Text</h1>')
+  end
+
+  before do
+    bilbo
+    frodo
+    marketing
+    announcement
+  end
```
Comment (awead) [inline spec/models/content_block_spec.rb]: shouldn't need this if you use `let!`

Comment (mjgiarlo) [inline spec/models/content_block_spec.rb]: Done.

File: app/models/content_block.rb [hunk 3]
```diff
@@ -3,12 +3,38 @@ class ContentBlock < ActiveRecord::Base
   RESEARCHER = 'featured_researcher'.freeze
   ANNOUNCEMENT = 'announcement_text'.freeze
 
+  def self.marketing_text
+    find_or_create_by(name: MARKETING)
+  end
+
+  def self.marketing_text=(value)
+    marketing_text.tap do |marketing|
```
Comment (cbeer) [inline app/models/content_block.rb]: Can this just be `marketing_text.update(value: value)`?

File: app/models/content_block.rb [hunk 4]
```diff
@@ -3,12 +3,38 @@ class ContentBlock < ActiveRecord::Base
...
+    end
+  end
+
   def self.recent_researchers
     where(name: RESEARCHER).order('created_at DESC')
   end
 
   def self.featured_researcher
-    recent_researchers.first
+    recent_researchers.first || create(name: RESEARCHER)
```
Comment (cbeer) [inline app/models/content_block.rb]: Does `#first_or_create` work here?

**Comment by awead** [comment]
Travis is 😞

**Comment by mjgiarlo** [comment]
@awead oops, :cop:. Pushing a fix in a minute.

File: app/models/content_block.rb (hunk 3 above)
Comment (mjgiarlo) [inline app/models/content_block.rb]: :+1:

File: app/models/content_block.rb (hunk 4 above)
Comment (mjgiarlo) [inline app/models/content_block.rb]: yep!

**Comment by cbeer** [comment]
LGTM.

**Comment by awead** [comment]
👍

**Comment by mjgiarlo** [comment]
:shipit: ?

**Pull request merged**